src/libraries/System.Net.Http.Json/src/System/Net/Http/Json/JsonContent.cs

```diff
@@ -52,15 +52,15 @@ public static JsonContent Create(object? inputValue, Type inputType, MediaTypeHe
             => new JsonContent(inputValue, inputType, mediaType, options);
 
         protected override Task SerializeToStreamAsync(Stream stream, TransportContext? context)
-            => SerializeToStreamAsyncCore(stream, CancellationToken.None);
+            => SerializeToStreamAsyncCore(stream, async: true, CancellationToken.None);
 
         protected override bool TryComputeLength(out long length)
         {
             length = 0;
             return false;
         }
 
-        private async Task SerializeToStreamAsyncCore(Stream targetStream, CancellationToken cancellationToken)
+        private async Task SerializeToStreamAsyncCore(Stream targetStream, bool async, CancellationToken cancellationToken)
```

Comment:
I know this is still an `async` method but given that it is no longer used only by `SerializedToStreamAsync`, can we rename it to `SerializeToStreamCore`?

Reply:
It's called from all `SerializeToStreamAsync` and `SerializeToStream`:
https://github.com/dotnet/runtime/blob/bc77fbb3b6ff3c323d6f5498ea7c2922e698be94/src/libraries/System.Net.Http.Json/src/System/Net/Http/Json/JsonContent.cs#L55
https://github.com/dotnet/runtime/blob/bc77fbb3b6ff3c323d6f5498ea7c2922e698be94/src/libraries/System.Net.Http.Json/src/System/Net/Http/Json/JsonContent.netcoreapp.cs#L14
https://github.com/dotnet/runtime/blob/bc77fbb3b6ff3c323d6f5498ea7c2922e698be94/src/libraries/System.Net.Http.Json/src/System/Net/Http/Json/JsonContent.netcoreapp.cs#L17

Reply:
The param `async` already indicates if the method will use async APIs or not, what's the point of keeping the "Async" in the method name?

Reply:
Convention and consistency.